### PR TITLE
fix(trade-ideas): lock budget-log appends and fail closed on broken history

### PR DESCRIPTION
**Pull request by Solders-Girdles**
## Summary

Hardens `RiskBudgetLog` against the two gaps it had versus its sibling `AutonomyStateLog` (raised in the PR #1185 review discussion: the CLI `ideas budget set` can race the web console's budget POST, which only holds an in-process lock).

- **Locked appends.** `append()` did an unlocked read-then-append, so two processes could both append the same next version. It now holds an OS-level `FileLock` (`risk_budget.jsonl.lock`, same `filelock` dependency the paper cycle already uses) around the version check + write. The losing writer gets a clean `BudgetIntegrityError` and must re-read the budget it is renegotiating against; a lock timeout (10s) also surfaces as `BudgetIntegrityError`.
- **Validated reads.** `history()` now raises `BudgetIntegrityError` on malformed lines, non-contiguous (including duplicated) versions, and unreadable files, mirroring `AutonomyStateLog.history()`.

## Fail-closed decision

The autonomy log fails closed to `research_only`; the budget log has no safe analog because the seeded `DEFAULT_RISK_BUDGET` may be *wider* than the operator's current version and lacks the operator-attested `account_equity`. So budget resolution fails closed by **propagating** the integrity error: `peek_budget()` / `current_budget()` / gate paths refuse to resolve a budget from a broken log rather than silently falling back to defaults. The CLI already maps `BudgetIntegrityError` to `OPERATION_FAILED`, preflight reports "risk budget unreadable", and the Stage-2 cycle records a failed manifest row.

One behavioral consequence handled: `TradeIdeaService.current_budget()` seeding on first use can now lose the seed race cleanly (previously it would have silently duplicated version 1); it adopts the concurrent winner's entry instead of erroring.

Also folds in two small consistency fixes: preflight's `_validate_budget_history` contiguity loop was made redundant by the read-side validation, and the "unlike the budget log" note in `autonomy.py` is now stale.

## Testing

- New unit tests: malformed line / malformed decimal / version gap / duplicated version (the unlocked-race artifact) all raise `BudgetIntegrityError`; append times out with a foreign lock held and succeeds after release; service adopts a concurrent seed; service budget resolution fails closed on a corrupt log.
- `uv run pytest tests/unit -n auto -q` — 6572 passed.
- `uv run ruff check . --fix`, `uv run black .`, `uv run mypy src/gpt_trader`, `uv run agent-naming`, `uv run agent-regenerate` — all clean (regenerated var/agents artifacts are line-number shifts only).

## Summary by CodeRabbit

- **Bug Fixes**
  - Strengthened trade-idea risk budget log integrity checks for malformed, unreadable, corrupt numeric fields, truncated/tampered, and non-sequential/duplicate entries.
  - Improved concurrency-safe risk budget seeding so only the winning version is adopted after contention.
  - Hardened lock contention and unusable lock-file handling to fail safely instead of masking corruption.
  - Updated preflight “readiness” to validate both the budget log and its sidecar lock.
- **Tests**
  - Expanded unit coverage for corrupted logs, version gaps/duplicates, concurrency/race scenarios, and lock edge cases.
- **Chores**
  - Updated environment-variable and test metadata counts.

### CODE REVIEW

File: src/gpt_trader/features/trade_ideas/budget.py [hunk 1]
```diff
@@ -184,38 +186,94 @@ def from_dict(cls, payload: dict[str, Any]) -> BudgetLogEntry:
...
-                field="version",
-                value=entry.budget.version,
-            )
         self._path.parent.mkdir(parents=True, exist_ok=True)
-        line = json.dumps(entry.to_dict(), sort_keys=True, separators=(",", ":"))
-        with self._path.open("a", encoding="utf-8") as handle:
-            handle.write(line + "\n")
+        try:
+            self._lock.acquire(timeout=self._LOCK_TIMEOUT_SECONDS)
+        except Timeout as error:
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/budget.py:217]: Fixed in 646c7c05: `OSError` from lock acquisition is now wrapped as `BudgetIntegrityError` ("Budget log lock is unusable"), with a regression test covering a directory squatting on the lock path.

File: src/gpt_trader/features/trade_ideas/budget.py [hunk 2]
```diff
@@ -184,38 +186,94 @@ def from_dict(cls, payload: dict[str, Any]) -> BudgetLogEntry:
...
+                            value=entry.budget.version,
+                        )
+                    entries.append(entry)
+        except OSError as error:
+            raise BudgetIntegrityError(
+                f"Budget log is unreadable: {error}",
+                field="path",
+                value=str(self._path),
+            ) from error
         return entries
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/budget.py:283]: Deliberately mirroring `AutonomyStateLog`: reads stay lock-free. A reader that races an in-flight append can only observe a prefix of the final single-line `write()` (append-mode, one line per version), and the new read-side validation fails closed with `BudgetIntegrityError` on any torn/malformed line rather than acting on it — the same posture the autonomy log shipped with. Taking the lock in `history()` would also put every read path (web console renders, preflight, gate checks) in contention with writers for no correctness gain given the fail-closed reads.

File: src/gpt_trader/features/trade_ideas/budget.py [hunk 3]
```diff
@@ -184,38 +186,100 @@ def from_dict(cls, payload: dict[str, Any]) -> BudgetLogEntry:
...
             return []
         entries: list[BudgetLogEntry] = []
-        with self._path.open("r", encoding="utf-8") as handle:
-            for line in handle:
-                line = line.strip()
-                if line:
-                    entries.append(BudgetLogEntry.from_dict(json.loads(line)))
+        try:
+            with self._path.open("r", encoding="utf-8") as handle:
+                for line_number, raw_line in enumerate(handle, start=1):
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/budget.py:250]: Fixed in aee4f79a: `UnicodeDecodeError` is now wrapped alongside `OSError` as "Budget log is unreadable", with a regression test writing invalid UTF-8 bytes.

File: src/gpt_trader/features/trade_ideas/budget.py [hunk 4]
```diff
@@ -184,38 +186,100 @@ def from_dict(cls, payload: dict[str, Any]) -> BudgetLogEntry:
...
+    seeded defaults may be wider (and lack the operator-attested equity) than
+    the operator's current version, so a tampered or duplicated log must stop
+    budget resolution rather than fall back.
+    """
+
+    _LOCK_TIMEOUT_SECONDS = 10.0
 
     def __init__(self, path: Path) -> None:
         self._path = path
+        self._lock = FileLock(str(path) + ".lock")
```
[the remark this answers is not in the document]
Comment (Solders-Girdles) [reply src/gpt_trader/features/trade_ideas/budget.py:207]: Fixed in 3f388227: readiness now runs the same appendability check on `risk_budget.jsonl.lock` (label "risk budget lock"), with a test squatting a directory on the lock path.
